sdk/data/aztables/README.md

```diff
@@ -1,151 +1,380 @@
+        "OnSale": true,
+        "ReducedPrice": 7.99,
+        "PurchaseDate": aztables.EDMDateTime(time.Date(2021, time.August, 21, 1, 1, 0, 0, time.UTC)),
+        "BinaryRepresentation": aztables.EDMBinary([]byte{"Bytesliceinfo"})
+    }
+}
+marshalled, err := json.Marshal(myEntity)
+handle(err)
+
+client, err := service.NewClient("myTable")
+handle(err)
+
+resp, err := client.AddEntity(context.Background(), marshalled, nil)
+handle(err)
+```
+
+### Querying entities
+Querying entities in the table:
+
+```golang
+cred, err := aztables.NewSharedKeyCredential("myAccountName", "myAccountKey")
+handle(err)
+client, err := aztables.NewClient("https://myAccountName.table.core.windows.net/myTableName", cred, nil)
+handle(err)
+
+filter := "PartitionKey eq 'markers' or RowKey eq 'id-001'"
+options := &ListEntitiesOptions{
+    Filter: &filter,
+    Select: to.StringPtr("RowKey,Value,Product,Available"),
+    Top: to.Int32Ptr(15),
+}
+
+pager := client.List(options)
+for pager.NextPage(context.Background()) {
+    resp := pager.PageResponse()
+    fmt.Printf("Received: %v entities\n", len(resp.Entities))
+
+    for _, entity := range resp.Entities {
+        var myEntity aztables.EDMEntity
+        err = json.Unmarshal(entity, &myEntity)
+        handle(err)
+
+        fmt.Printf("Received: %v, %v, %v, %v\n", myEntity.Properties["RowKey"], myEntity.Properties["Value"], myEntity.Properties["Product"], myEntity.Properties["Available"])
+    }
+}
+
+err := pager.Err()
+handle(err)
+```
 
-- `TableServiceClient` - Client that provides methods to interact at the Table Service level such as creating, listing, and deleting tables
-- `TableClient` - Client that provides methods to interact at an table entity level such as creating, querying, and deleting entities within a table.
-- `Table` - Tables store data as collections of entities.
-- `Entity` - Entities are similar to rows. An entity has a primary key and a set of properties. A property is a name value pair, similar to a column.
-
-Common uses of the Table service include:
-
-- Storing TBs of structured data capable of serving web scale applications
-- Storing datasets that don't require complex joins, foreign keys, or stored procedures and can be de-normalized for fast access
-- Quickly querying data using a clustered index
-### Create the Table service client
-
-First, we need to construct a `TableServiceClient`.
-
-### Create an Azure table
-Next, we can create a new table.
-
-
-### Get an Azure table
-The set of existing Azure tables can be queries using an OData filter.
-
+## Troubleshooting
 
-### Delete an Azure table
+### Error Handling
 
-Individual tables can be deleted from the service.
+All I/O operations will return an `error` that can be investigated to discover more information about the error. In addition, you can investigate the raw response of any response object:
+```golang
+resp, err := client.CreateTable(context.Background(), nil)
+if err != nil {
+    fmt.Println(resp.RawResponse)
```

Comment:
This shouldn't work, and generally speaking, accessing the "thing" when `err != nil` is an anti-pattern.

To get the raw response, you need to use `errors.As()` on the `err` to convert it to an [`azcore.HTTPResponse`](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/errors.go#L21) interface which provides access to it.